internal/bminventory/inventory.go

```diff
@@ -132,7 +132,7 @@ type InstallerInternals interface {
 
 //go:generate mockgen -package bminventory -destination mock_crd_utils.go . CRDUtils
 type CRDUtils interface {
-	CreateAgentCR(ctx context.Context, log logrus.FieldLogger, hostId, clusterNamespace, clusterName string) error
+	CreateAgentCR(ctx context.Context, log logrus.FieldLogger, hostApi host.API, hostId, clusterNamespace, clusterName string) error
```

Comment:
If object use some external interface it's common to give in the builder, for example in `NewCRDUtils`

Reply:
sure, done